Sources/System/Sockets/SocketDescriptor+Messages.swift

```diff
@@ -0,0 +1,453 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+// @available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *)
+extension SocketDescriptor {
+  /// A reusable collection of variable-sized ancillary messages
+  /// sent or received over a socket. These represent protocol control
+  /// related messages or other miscellaneous ancillary data.
+  ///
+  /// Corresponds to a buffer of `struct cmsghdr` messages in C, as used
+  /// by `sendmsg` and `recmsg`.
+  public struct AncillaryMessageBuffer {
```

Comment:
Because that's not what `sendmsg`/`recvmsg` expects. This type models the flat buffer of variable-sized control messages, exactly as expected by these two functions. If we modeled it with an array, then we'd have to allocate a buffer within send/recv to encode it -- which wouldn't be viable.

The socket library provides iterator macros to walk the control message buffer & access the individual elements. This type provides the same abstraction in the form of a forward Collection. (Hm, should it just be Sequence?)

Reply:
Is it layout compatible with the C array?

Reply:
I see, we're trying to make a type that is layout compatible with the `void *msg_control`, which is fully flattened. I'll re-read this with that in mind.